dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/ClientIT.java

```diff
@@ -300,112 +348,203 @@ private void assertDepsCommandOutput() {
+        checkCommandForHelp(new String[] { WORKFLOW, "restub" });
+
+        checkCommandForHelp(new String[] { WORKFLOW, DOWNLOAD, HELP });
+        checkCommandForHelp(new String[] { WORKFLOW, LIST, HELP });
+        checkCommandForHelp(new String[] { WORKFLOW, SEARCH, HELP });
+        checkCommandForHelp(new String[] { WORKFLOW, PUBLISH, HELP });
+        checkCommandForHelp(new String[] { WORKFLOW, INFO, HELP });
+        checkCommandForHelp(new String[] { WORKFLOW, "cwl", HELP });
+        checkCommandForHelp(new String[] { WORKFLOW, "wdl", HELP });
+        checkCommandForHelp(new String[] { WORKFLOW, REFRESH, HELP });
+        checkCommandForHelp(new String[] { WORKFLOW, LABEL, HELP });
+        checkCommandForHelp(new String[] { WORKFLOW, CONVERT, HELP });
+        checkCommandForHelp(new String[] { WORKFLOW, CONVERT, CWL_2_JSON, HELP });
+        checkCommandForHelp(new String[] { WORKFLOW, CONVERT, CWL_2_YAML, HELP });
+        checkCommandForHelp(new String[] { WORKFLOW, CONVERT, "wd2json", HELP });
+        checkCommandForHelp(new String[] { WORKFLOW, CONVERT, ENTRY_2_JSON, HELP });
+        checkCommandForHelp(new String[] { WORKFLOW, LAUNCH, HELP });
+        checkCommandForHelp(new String[] { WORKFLOW, VERSION_TAG, HELP });
+        checkCommandForHelp(new String[] { WORKFLOW, UPDATE_WORKFLOW, HELP });
+        checkCommandForHelp(new String[] { WORKFLOW, MANUAL_PUBLISH, HELP });
+        checkCommandForHelp(new String[] { WORKFLOW, "restub", HELP });
+        checkCommandForHelp(new String[] { WORKFLOW, STAR, HELP });
+        checkCommandForHelp(new String[] { WORKFLOW, TEST_PARAMETER, HELP });
+        checkCommandForHelp(new String[] { WORKFLOW, VERIFY, HELP });
+        checkCommandForHelp(new String[] { WORKFLOW });
+
+        checkCommandForHelp(new String[] { PLUGIN, LIST, HELP });
+        checkCommandForHelp(new String[] { PLUGIN, DOWNLOAD, HELP });
+        checkCommandForHelp(new String[] { PLUGIN });
+
+        checkCommandForHelp(new String[] { CHECKER });
+        checkCommandForHelp(new String[] { CHECKER, DOWNLOAD, HELP });
+        checkCommandForHelp(new String[] { CHECKER, LAUNCH, HELP });
+        checkCommandForHelp(new String[] { CHECKER, ADD, HELP });
+        checkCommandForHelp(new String[] { CHECKER, UPDATE, HELP });
+        checkCommandForHelp(new String[] { CHECKER, UPDATE_VERSION, HELP });
+        checkCommandForHelp(new String[] { CHECKER, TEST_PARAMETER, HELP });
+
     }
 
     private void checkCommandForHelp(String[] argv) throws IOException {
         final ArrayList<String> strings = Lists.newArrayList(argv);
-        strings.add("--config");
+        strings.add(CONFIG);
         strings.add(TestUtility.getConfigFileLocation(true));
 
         Client.main(strings.toArray(new String[0]));
         assertTrue(systemOutRule.getText().contains("Usage: dockstore"));
         systemOutRule.clear();
     }
 
+
+    @Test
+    public void touchAllInvalidCommands() throws IOException {
```

Comment:
Some nitpicking. If I understand correctly, this isn't testing all possible invalid commands (which is infinite) but it does test all valid commands by changing them slightly. So maybe `testAllValidCommandsWithSlightPermutation` for example

Reply:
Yeah, that's a better name. I have changed it to the one you suggested.